frontend/shared-components/data-table/data-table.tsx

```diff
@@ -0,0 +1,178 @@
+        full: faGreaterThan,
+        save: faPlusCircle,
+      },
+      {
+        dates: "2003/01/01",
+        incidentType: "Use of force",
+        officersInvolved: ["John Smith"],
+        subject: "unknown",
+        source: "News Article",
+        full: faGreaterThan,
+        save: faPlusCircle,
+      },
+      {
+        dates: "2003/01/01",
+        incidentType: "Use of force",
+        officersInvolved: ["John Smith"],
+        subject: "unknown",
+        source: "News Article",
+        full: faGreaterThan,
+        save: faPlusCircle,
+      },
+      {
+        dates: "2003/01/01",
+        incidentType: "Use of force",
+        officersInvolved: ["John Smith"],
+        subject: "unknown",
+        source: "News Article",
+        full: faGreaterThan,
+        save: faPlusCircle,
+      },
+      {
+        dates: "2003/01/01",
+        incidentType: "Use of force",
+        officersInvolved: ["John Smith"],
+        subject: "unknown",
+        source: "News Article",
+        full: faGreaterThan,
+        save: faPlusCircle,
+      },
+      {
+        dates: "2003/01/01",
+        incidentType: "Use of force",
+        officersInvolved: ["John Smith"],
+        subject: "unknown",
+        source: "News Article",
+        full: faGreaterThan,
+        save: faPlusCircle,
+      },
+    ],
+    []
+  )
+
+  const columns = React.useMemo(
+    () => [
+      {
+        Header: "Date(s)",
+        accessor: "dates" as const,
+      },
+      {
+        Header: "Incident Type",
+        accessor: "incidentType" as const,
+      },
+      {
+        Header: "Officer(s) Involved",
+        accessor: "officersInvolved" as const,
+      },
+      {
+        Header: "Subject",
+        accessor: "subject" as const,
+      },
+      {
+        Header: "Source",
+        accessor: "source" as const,
+      },
+      {
+        Header: "Full Record",
+        accessor: "full" as const,
+      },
+      {
+        Header: "Save Record",
+        accessor: "save" as const,
+      },
+    ],
+    []
+  )
+  const tableInstance = useTable({ columns, data })
+
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = tableInstance
+
+  return (
+    <div className={dTableWrapper}>
+      <table {...getTableProps()} className={dTable}>
+        <thead className={dHeader}>
+          {headerGroups.map((headerGroup) => (
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((column) => (
+                <th {...column.getHeaderProps()}>{column.render("Header")}</th>
+              ))}
+            </tr>
+          ))}
+        </thead>
+        <tbody {...getTableBodyProps()}>
+          {rows.map((row) => {
+            prepareRow(row)
+            return (
+              <tr {...row.getRowProps()}>
+                {row.cells.map((cell) => {
+                  console.log(cell)
+                  if (cell.column.id === "full" || cell.column.id === "save") {
```

Comment:
We could deconstruct the `id` directly from the parameter to avoid writing its reference over again!
Also, the lines 150-154 can probably appropriately be a one-liner! 

Generally, separation of concerns-wise we probably want to keep the amount of logic we perform directly in the template code pretty minimal - I've always gone by the rule of thumb that if it extends past 2-3 lines it should be factored out or simplified

Reply:
I'm definitely open to suggestions as to how to clean this up, this is the implementation detail I'm unhappiest with here.